lib/wallet/rpc.js

```diff
@@ -1586,6 +1587,40 @@ class RPC extends RPCBase {
     return sig.toString('base64');
   }
 
+  async signMessageWithName(args, help) {
+    if (help || args.length !== 2) {
+      throw new RPCError(errs.MISC_ERROR,
+        'signmessagewithname "name" "message"');
+    }
+
+    const wallet = this.wallet;
+    const valid = new Validator(args);
+    const name = valid.str(0, '');
+    const str = valid.str(1, '');
+
+    if (!name || !rules.verifyName(name))
+      throw new RPCError(errs.TYPE_ERROR, 'Invalid name.');
+
+    if (!wallet.master.key)
+      throw new RPCError(errs.WALLET_UNLOCK_NEEDED, 'Wallet is locked.');
```

Comment:
I don't think we need this here, it will be checked in `signMessage()` and we don't need to decrypt the wallet until then anyway.